aptos-move/aptos-vm-types/src/module_and_script_storage/state_view_adapter.rs

```diff
@@ -72,6 +82,62 @@ impl<'s, S: StateView, E: WithRuntimeEnvironment> AptosCodeStorageAdapter<'s, S,
         let storage = adapter.into_unsync_code_storage(runtime_environment);
         Self { storage }
     }
+
+    /// Drains cached verified modules from the code storage, transforming them into format used by
+    /// global caches. Should only be called when the code storage borrows [StateView].
+    pub fn into_verified_module_code_iter(
+        self,
+    ) -> Result<
+        impl Iterator<
+            Item = (
+                ModuleId,
+                Arc<ModuleCode<CompiledModule, Module, AptosModuleExtension>>,
+            ),
+        >,
+        PanicError,
+    > {
+        let state_view = match self.storage.module_storage().byte_storage().state_view {
+            BorrowedOrOwned::Borrowed(state_view) => state_view,
+            BorrowedOrOwned::Owned(_) => {
+                return Err(PanicError::CodeInvariantError(
+                    "Verified modules should only be extracted from borrowed state".to_string(),
+                ))
+            },
+        };
+
+        let mut modules_to_add = vec![];
+        for (key, verified_code) in self
+            .storage
+            .into_module_storage()
+            .into_verified_modules_iter()
+        {
+            // We have cached the module previously, so we must be able to find it in storage.
+            let extension = state_view
+                .get_state_value(&StateKey::module_id(&key))
+                .map_err(|err| {
+                    let msg = format!(
+                        "Failed to retrieve module {}::{} from storage {:?}",
+                        key.address(),
+                        key.name(),
+                        err
+                    );
+                    PanicError::CodeInvariantError(msg)
+                })?
+                .map(AptosModuleExtension::new)
+                .ok_or_else(|| {
```

Comment:
nit: map_or_else? or maybe it's preferred this way, does linter have an opinion?